scipy/interpolate/rbf.py

```diff
@@ -172,46 +337,105 @@ def _init_function(self, r):
                     self._function = self.function.__get__(self, Rbf)
                 else:
                     import new
-                    self._function = new.instancemethod(self.function, self,
-                                                        Rbf)
+                    self._function = new.instancemethod(self.function, self, Rbf)
             else:
-                raise ValueError("Function argument must take 1 or 2 arguments.")
-
-        a0 = self._function(r)
-        if a0.shape != r.shape:
+                raise ValueError("Function argument must take 1 or 2 arguments")
+
+    def _init_default_epsilon(self):
+        # default epsilon is the "the average distance between adjacent nodes"
+        ximax = np.amax(self.xi, axis=1)
+        ximin = np.amin(self.xi, axis=1)
+        self.epsilon = np.power(np.prod(ximax - ximin)/self.N, 1.0/self.dim)
+
+    def _interpolation_matrix(self):
+        R = self._call_norm(self.xi, self.xi)
+        A = self._function(R)
+        if R.shape != A.shape:
             raise ValueError("Callable must take array and return array of the same shape")
-        return a0
+        if self.smooth != 0.0:
+            eigmax = linalg.eigh(A, eigvals_only=True, eigvals=(self.N-1, self.N-1))[0]
```

Comment:
I think this needs still some thought.

It needs to respect `sign`. For user-defined functions, you'd probably need to calculate both ends of spectrum.
The meaning of the `smooth` keyword is also drastically changed --- the name should probably be changed too.

One question is also what is a good definition for the smooth parameter --- it might be good to make a connection to `rcond` used in pseudoinverses.